grype/pkg/syft_provider.go

```diff
@@ -26,8 +26,15 @@ func syftProvider(userInput string, config ProviderConfig) ([]Package, Context,
 	if err != nil {
 		return nil, Context{}, nil, err
 	}
+	s := &sbom.SBOM{
+		Artifacts: sbom.Artifacts{
+			Packages:          catalog,
+			LinuxDistribution: theDistro,
+		},
+		Relationships: relationships,
+	}
 
-	catalog = removePackagesByOverlap(catalog, relationships)
+	catalog = removePackagesByOverlap(s)
```

Comment:
nit: what do you think of the return type here? we are passing in an sbom `s` only for `sbom` to be remade with the modified catalog on line 47 down below - should this instead be a method on `s` so we can reuse it and not have to  rebuild it later down the provider?

Reply:
Thanks for catching this. I didn't notice that we were re-assembling another SBOM struct below. I'll see if this can be easily rearranged.